lib/rule-data-snapshot.js

```diff
@@ -3,6 +3,14 @@
 
 var merge = require('lodash.mergewith');
 
+function isEmptyObj (obj) {
+  return obj && (typeof obj == "object") && Object.keys(obj).length === 0 && obj.constructor === Object
```

Comment:
Missing semicolon.

Replace double quote for single for consistency.

Why do you need to test the constructor?